suffuse/error.go

```diff
@@ -0,0 +1,70 @@
+package suffuse
+
+import (
+  "errors"
+  "reflect"
+)
+
+/** If you have a *Foo type which implements error and you pass it
+ *  to a method accepting error, when you pass a nil pointer it
+ *  does NOT test as nil. It's apparently a non-nil interface with
+ *  a nil value. That's what the second test is for.
+ */
+func IsNilError(e error)bool {
+  return e == nil || reflect.ValueOf(e).IsNil()
```

Comment:
Why not? Reflection works. "Let's manually make sure" never does. Speed? I'm not ready to trade correctness for performance at this stage of the game. Is there an enforceable way to avoid this problem? There's a go linter I guess, maybe it catches this?